internal/osutils/autostart/autostart_lin.go

```diff
@@ -2,17 +2,127 @@
 
 package autostart
 
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/gobuffalo/packr"
+	"github.com/mitchellh/go-homedir"
+)
+
 func (a *App) Enable() error {
-	panic("Not implemented")
+	appPath, err := ensuredAppPath()
```

Comment:
`ensuredAppPath` is writing a bunch of files, but you are calling it before even checking if we're already enabled.

Granted `ensuredAppPath` doesn't write when the main app file already exists, but that logic of "don't do this if it already exists" is exactly what should be happening in this function. Your `ensuredAppPath` function should just be in charge of creating the files, not worrying about whether it should.